### DISCUSSION TITLE
SmartShift as a Service (SSaaS)

### DISCUSSION THREAD

**Discussion by rohrsh** [amberelectric/public-api#94] [Home Assistant]
I was briefly part of the SmartShift trial with a device to control my pool pump. It wasn't an attractive solution for me though, since my pool controller is a smart device that provides data 24/7 on temperature and chemistry so I don't want to shut off it's power. It is however a smart pool controller than I am able to remotely control via Modbus. 

Like many, I'm wrestling with how to run that pool pump to take advantage of low prices during the day. It's not a trivial exercise, perhaps not able to be done well with Home Assistant automations. 

What I would like to do is to take all of the "smarts" from SmartShift and use it to control my pool pump via software. 

e.g. a service I can subscribe to that tells me when to run. I would need to define some parameters, such as my required hours per day, service windows, etc. 

I was tempted to keep my SmartShift device and just read its state (it's a Belkin device) and use that as an input to my automations. But I think the SS team got grumpy when they saw no load attached. 

Anyhow. SmartShift as a Service. Bring it on.

**Comment by madpilot** [comment]
This has been discussed. Nothing concrete yet, but it'll be possible in the future.

**Comment by andymcblane** [comment]
I'd also be keen on this, it's a difficult problem to solve especially when the price is fluctuating (wonder what summer will look like?)

**Comment by rohrsh** [comment]
Best concept I've seen so far: https://github.com/davidusb-geek/emhass

**Comment by purcell-lab** [comment]
If you can start/ stop your pool pump via ModBus then HomeAssistant can easily control based off the Amber price but also your other pool variables. Ie if chemistry is way off run your pump during the cheapest 8 hours otherwise run your pump during the cheapest 4 hours a day.

A straight forward approach is to switch your devices on when prices are low (green) and switch them off when prices are high (yellow) or very high (red). I created a set of automations to do exactly that:
https://community.home-assistant.io/t/running-devices-when-energy-is-cheaper-and-greener/380011?u=markpurcell

I am now running a more complex approach using [EMHASS](https://emhass.readthedocs.io/en/latest/intro.html)  which gives me even more optimisation of when to schedule my devices.
![Screenshot_20221022-063656](https://user-images.githubusercontent.com/79175134/197284495-ccaeecf2-a5fb-4df1-8642-b653542db0b9.png)
